src/SfxWeb/src/app/modules/event-store/event-store/event-store.component.ts

```diff
@@ -53,97 +62,117 @@ export class EventStoreComponent implements OnInit, OnDestroy {
   }
 
   public setDate(date: IQuickDates) {
       this.setNewDates({
-        endDate: new Date(this.eventsList.endDate),
-        startDate: TimeUtils.AddHours(this.endDate, -1 * date.hours)
+          endDate: new Date(this.listEventStoreData[0].eventsList.endDate),
+          startDate: TimeUtils.AddHours(this.endDate, -1 * date.hours)
       });
   }
 
   private setNewDateWindow(): void {
-      if (this.eventsList.setDateWindow(this.startDate, this.endDate)) {
+      const subs = this.listEventStoreData.filter(data => data.eventsList.setDateWindow(this.startDate, this.endDate))
+      .map((data) => {
           this.resetSelectionProperties();
           this.isResetEnabled = true;
-          this.eventsList.reload().subscribe( data => {
-              this.setTimelineData();
-          });
-      } else {
-          this.resetSelectionProperties();
+          return data.eventsList.reload();
+      });
+
+      if (subs.length > 0) {
+          forkJoin(subs).subscribe(() => this.setTimelineData());
       }
   }
 
+  public mergeTimelineData(combinedData: ITimelineData, data: ITimelineData): void {
+      data.items.forEach(item => combinedData.items.add(item));
+
+      data.groups.forEach(group => combinedData.groups.add(group));
+
+      combinedData.potentiallyMissingEvents =
+          combinedData.potentiallyMissingEvents || data.potentiallyMissingEvents;
+  }
+
+  private initializeTimelineData(): ITimelineData {
+      return {
+          start: this.startDate,
+          end: this.endDate,
+          groups: new DataSet<DataGroup>(),
+          items: new DataSet<DataItem>()
+      };
+  }
+
   public setTimelineData(): void {
-    this.eventsList.ensureInitialized().subscribe( () => {
-        try {
-            if (this.pshowAllEvents) {
-                const d = parseEventsGenerically(this.eventsList.collection.map(event => event.raw), this.transformText);
-
-                this.timeLineEventsData = {
-                    groups: d.groups,
-                    items: d.items,
-                    start: this.startDate,
-                    end: this.endDate,
-                    potentiallyMissingEvents: d.potentiallyMissingEvents
-                };
-
-            }else if (this.timelineGenerator) {
-                const d = this.timelineGenerator.generateTimeLineData(this.eventsList.collection.map(event => event.raw), this.startDate, this.endDate);
-
-                this.timeLineEventsData = {
-                    groups: d.groups,
-                    items: d.items,
-                    start: this.startDate,
-                    end: this.endDate,
-                    potentiallyMissingEvents: d.potentiallyMissingEvents
-                };
-            }
-        }catch (e) {
-            console.error(e);
-        }
-    });
-    }
-
-    setNewDates(dates: IOnDateChange) {
-        this.debounceHandler.next(dates);
-    }
+      let rawEventlist = [];
```

Comment:
move these into the subscribe callback to keep them as close to their scope as they can.